fmc/src/flow/x509.rs

```diff
@@ -0,0 +1,83 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    x509.rs
+
+Abstract:
+
+    File contains X509 Certificate & CSR related utility functions
+
+--*/
+use super::crypto::Crypto;
+use crate::fmc_env::FmcEnv;
+use caliptra_drivers::*;
+
+/// X509 API
+pub enum X509 {}
+
+impl X509 {
+    /// Get X509 Subject Serial Number
+    ///
+    /// # Arguments
+    ///
+    /// * `env`     - FMC Environment
+    /// * `pub_key` - Public Key
+    ///
+    /// # Returns
+    ///
+    /// `[u8; 64]` - X509 Subject Identifier serial number
+    pub fn subj_sn(env: &mut FmcEnv, pub_key: &Ecc384PubKey) -> CaliptraResult<[u8; 64]> {
+        let data = pub_key.to_der();
+        let digest = Crypto::sha256_digest(env, &data);
+        let digest = okref(&digest)?;
+        Ok(Self::hex(&digest.into()))
+    }
+
+    /// Get Cert Subject Key Identifier
+    ///
+    /// # Arguments
+    ///
+    /// * `env`     - FMC Environment
+    /// * `pub_key` - Public Key
+    ///
+    /// # Returns
+    ///
+    /// `[u8; 20]` - X509 Subject Key Identifier
+    pub fn subj_key_id(env: &mut FmcEnv, pub_key: &Ecc384PubKey) -> CaliptraResult<[u8; 20]> {
+        let data = pub_key.to_der();
+        let digest = Crypto::sha256_digest(env, &data);
+        let digest: [u8; 32] = okref(&digest)?.into();
+        Ok(digest[..20].try_into().unwrap())
```

Comment:
I think you can do this without unwrap by doing

```rust
let digest = Crypto::sha256_digest(env, &data);
let out = [0u8; 20];
out.copy_from_slice(digest[..20]);
Ok(out)
```

I would expect this to have the same amount of copying anyway.

Reply:
Rust doesn`t like that...because sha256_digest returns a Array4x8. I did something similar to avoid the last unwrap

 ```
       let digest = Crypto::sha256_digest(env, &data);
        let digest: [u8; 32] = okref(&digest)?.into();
        let mut out = [0u8; 20];
        out.copy_from_slice(&digest[..20]);
        Ok(out)
```